File: university-service/src/test/java/ua/com/foxminded/courseproject/validation/ClassRoomDtoTest.java
package ua.com.foxminded.courseproject.validation;

import org.junit.jupiter.api.Test;
import ua.com.foxminded.courseproject.dto.ClassRoomDto;

import javax.validation.ConstraintViolation;
import java.util.Set;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ClassRoomDtoTest extends ValidationSetupTest {

    @Test
    public void numberValidation_shouldThrowException_whenNumberIsNull() {
        ClassRoomDto classRoomDto = new ClassRoomDto();
        classRoomDto.setNumber(null);
        String expectedMessage = "must not be null";
        Integer expectedSize = 1;

        Set<ConstraintViolation<ClassRoomDto>> constraintViolations = validator.validate(classRoomDto);
        Integer actualSize = constraintViolations.size();
        String actualMessage = constraintViolations.iterator().next().getMessage();

        assertEquals(expectedMessage, actualMessage);
        assertEquals(expectedSize, actualSize);
    }

    @Test
    public void numberValidation_shouldThrowException_whenNumberIsNegative() {
        ClassRoomDto classRoomDto = new ClassRoomDto();
        classRoomDto.setNumber(-5);
        String expectedMessage = "Classroom number must be positive or equal 0.";
        Integer expectedSize = 1;

        Set<ConstraintViolation<ClassRoomDto>> constraintViolations = validator.validate(classRoomDto);
        Integer actualSize = constraintViolations.size();
        String actualMessage = constraintViolations.iterator().next().getMessage();

        assertEquals(expectedMessage, actualMessage);
        assertEquals(expectedSize, actualSize);
    }

    @Test
    public void numberValidation_shouldNotThrowException_whenNumberEqualsZero() {
        ClassRoomDto classRoomDto = new ClassRoomDto();
        classRoomDto.setNumber(0);
        Integer expectedSize = 0;

        Set<ConstraintViolation<ClassRoomDto>> constraintViolations = validator.validate(classRoomDto);
        Integer actualSize = constraintViolations.size();

        assertEquals(expectedSize, actualSize);
    }

    @Test
    public void numberValidation_shouldNotThrowException_whenNumberIsPositive() {
        ClassRoomDto classRoomDto = new ClassRoomDto();
        classRoomDto.setNumber(1);
        Integer expectedSize = 0;

        Set<ConstraintViolation<ClassRoomDto>> constraintViolations = validator.validate(classRoomDto);
        Integer actualSize = constraintViolations.size();

        assertEquals(expectedSize, actualSize);
    }


}

File: university-service/src/test/kotlin/ua/com/foxminded/courseproject/validation/ClassRoomDtoTest.kt
package ua.com.foxminded.courseproject.validation

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import ua.com.foxminded.courseproject.dto.ClassRoomDto
import validation.ValidationTestFixture.Companion.validator

class ClassRoomDtoTest {
    @Test
    fun numberValidation_shouldThrowException_whenNumberIsNull() {
        val classRoomDto = ClassRoomDto()
        classRoomDto.number = null
        val expectedMessage = "must not be null"
        val expectedSize = 1

        val constraintViolations = validator.validate(classRoomDto)
        val actualSize = constraintViolations.size
        val actualMessage = constraintViolations.iterator().next().message

        Assertions.assertEquals(expectedMessage, actualMessage)
        Assertions.assertEquals(expectedSize, actualSize)
    }

    @Test
    fun numberValidation_shouldThrowException_whenNumberIsNegative() {
        val classRoomDto = ClassRoomDto()
        classRoomDto.number = -5
        val expectedMessage = "Classroom number must be positive or equal 0."
        val expectedSize = 1

        val constraintViolations = validator.validate(classRoomDto)
        val actualSize = constraintViolations.size
        val actualMessage = constraintViolations.iterator().next().message

        Assertions.assertEquals(expectedMessage, actualMessage)
        Assertions.assertEquals(expectedSize, actualSize)
    }

    @Test
    fun numberValidation_shouldNotThrowException_whenNumberEqualsZero() {
        val classRoomDto = ClassRoomDto()
        classRoomDto.number = 0
        val expectedSize = 0

        val constraintViolations = validator.validate(classRoomDto)
        val actualSize = constraintViolations.size

        Assertions.assertEquals(expectedSize, actualSize)
    }

    @Test
    fun numberValidation_shouldNotThrowException_whenNumberIsPositive() {
        val classRoomDto = ClassRoomDto()
        classRoomDto.number = 1
        val expectedSize = 0

        val constraintViolations = validator.validate(classRoomDto)
        val actualSize = constraintViolations.size

        Assertions.assertEquals(expectedSize, actualSize)
    }
}
